samples/nrf9160/mqtt_simple/src/main.c

```diff
@@ -354,18 +382,33 @@ void main(void)
 	}
 
 	while (1) {
-		err = mqtt_input(&client);
-		if (err != 0) {
-			printk("ERROR: mqtt_input %d\n", err);
+		err = poll(&fds, 1, K_SECONDS(CONFIG_MQTT_KEEPALIVE));
+		if (err < 0) {
+			printk("ERROR: poll %d\n", errno);
+			continue;
+		} else if (err == 0) {
+			printk("poll() timed out, sending ping if needed...\n");
+
+			err = mqtt_live(&client);
```

Comment:
I'd say we should call `mqtt_live` not only when `poll` times-out, but in each iteration. It will keep track on when should it send the ping on its own, so no unneeded pings will be sent. And there's no guarantee, that we will actually send something when we receive data, so we might end up starving the connection.

Reply:
Will fix.

Reply:
Still not fixed, unless I miss something.